components/script/dom/bindings/str.rs

```diff
@@ -332,3 +346,74 @@ impl Extend<char> for DOMString {
         self.0.extend(iterable)
     }
 }
+
+/// https://html.spec.whatwg.org/multipage/#parse-a-month-component
+fn parse_month_component(value: &str) -> Result<(u32, u32), ()> {
+    // Step 3
+    let value_vec: Vec<&str> = value.split('-').collect();
```

Comment:
I'd like to avoid heap allocations as much as possible. The `collect()` call here heap allocates at least 2 strings when it parses a valid month component, which is quite a heavy overhead cost that we'd have to pay every time we try to validate month strings.

Reply:
@KiChjang  Could you tell me what `method` you want use here?

Reply:
Ok, I use another way